message_definitions/v1.0/common.xml

```diff
@@ -5319,6 +5319,17 @@
       <field type="uint32_t" name="active" display="bitmask">Active outputs</field>
       <field type="float[32]" name="actuator">Servo / motor output array values. Zero values indicate unused channels.</field>
     </message>
+    <message id="376" name="VISO_OBSTACLE_DISTANCE">
+      <description>Obstacle distances in front of the 3D depth camera, starting from the top left to bottom right divided in a 3X3 grid. This message assumes NEU frame of reference</description>
+      <field type="uint64_t" name="time_usec" units="us">Timestamp (UNIX Epoch time or time since system boot). The receiving end can infer timestamp format (since 1.1.1970 or since system boot) by checking for the magnitude of the number.</field>
+      <field type="uint8_t" name="sensor_type" enum="MAV_DISTANCE_SENSOR">Class id of the distance sensor type.</field>
+      <field type="uint16_t[9]" name="obstacle_ID"> Unique ID given to each obstacle so that its movement can be tracked. Use 0 if object ID is unknown.</field>
```

Comment:
normally in mavlink land it seems UINT16_MAX for unknown ids.

Reply:
we should probably add a "frame" field similar to other messages with a frame field.  AP will only implement the body frame for now but someone will bring this up I'm sure.

Reply:
Maybe we should call the message "VISION_OBSTACLE_DISTANCE"?

Reply:
Frame, Field of View (x/y) and Orientation (direction/offset of centre of FoV) should be definable (fields) I think.
Suggest name OBSTACLE_GRID_3X3